src/utils/validate.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/**
+ * Module dependencies.
+ */
+
+import { set } from 'lodash';
+import Ajv from 'ajv';
+
+/**
+ * Export `FieldErrorType` type.
+ */
+
+export type FieldErrorType = {|
+  args?: Object,
+  rule: string
+|};
+
+/**
+ * Get property name.
+ */
+
+function getPropertyName(error: Object): string {
+  const key = error.dataPath.substring(1);
+
+  switch (error.keyword) {
+    case 'required':
+      return `${key ? `${key}.` : ''}${error.params.missingProperty}`;
+
+    case 'additionalProperties':
+      return `${key ? `${key}.` : ''}${error.params.additionalProperty}`;
+
+    default:
+      return key;
+  }
+}
+
+/**
+ * Get error args.
+ */
+
+function getErrorArgs(error) {
+  switch (error.keyword) {
+    case 'maxLength':
+      return { max: error.params.limit };
+
+    case 'minLength':
+      return { min: error.params.limit };
+
+    default:
+      return;
+  }
+}
+
+/**
+ * Get error rule.
+ */
+
+function getErrorRule(error) {
+  switch (error.keyword) {
+    case 'additionalProperties':
```

Comment:
Why the need to rename this error type?
IMO it can still be an `additionalProperties` error type.
This way we don't change the `ajv` API errors.